osu.Game/Rulesets/RealmRulesetStore.cs

```diff
@@ -104,6 +111,25 @@ private void prepareDetachedRulesets(RealmAccess realmAccess)
             });
         }
 
+        private bool checkRulesetUpToDate(Ruleset instance)
+        {
+            switch (instance.RulesetAPIVersionSupported)
+            {
+                // The default `virtual` implementation leaves the version string empty.
+                // Consider rulesets which haven't override the version as up-to-date for now.
+                // At some point (once ruleset devs add versioning), we'll probably want to disallow this for deployed builds.
+                case @"":
+                // Rulesets which are bundled with the game. Saves having to update their versions each bump.
+                case @"internal":
```

Comment:
I can't help but point out that there's nothing stopping ruleset developers from specifying

```csharp
public virtual string RulesetAPIVersionSupported => "internal";
```

too and bypassing this versioning scheme entirely. They'd be *very very naughty* for doing this, but still.

That said I see the point of this escape hatch existing for strictly our purposes, and am unsure how to protect it from misuse. I guess the simplest way would be to disallow `internal` version spec when loading user rulesets (i.e. those from `$DOCS/osu/rulesets`)?

Reply:
We can do that if it is abused, definitely. I'd like to think ruleset developers have users' best interests in mind though.

Reply:
Hmm I'm actually not sure we need `internal`. It seems like rulesets should be able to do, verbatim:
```
public override string RulesetAPIVersionSupported => CURRENT_RULESET_API_VERSION;
```
Should just be able to work, since it looks to be compiled to a constant in IL:

https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABAzAAgwJhwGEcBvAWACgca9dIA7AZwBc8BGABhwEEcBeHACIeQgNxVadDtwBCAgHy8JlAL5A=

Which means we could probably change all of our internal rulesets to do this already, along with the templates, and any new ruleset authors wouldn't have to make any changes. It would only be the existing ruleset authors that would have to add that override themselves (and thus we would have to handle empty string).

I might need to test this to make sure...

Reply:
Just did some more testing and it looks like the above should work:

![image](https://user-images.githubusercontent.com/1329837/186933019-47dcc613-39dc-43b6-9f81-4c597c17acac.png)

Tested with:
```diff
diff --git a/Templates/Rulesets/ruleset-empty/osu.Game.Rulesets.EmptyFreeform/EmptyFreeformRuleset.cs b/Templates/Rulesets/ruleset-empty/osu.Game.Rulesets.EmptyFreeform/EmptyFreeformRuleset.cs
index 00754c6346..ffdf58925a 100644
--- a/Templates/Rulesets/ruleset-empty/osu.Game.Rulesets.EmptyFreeform/EmptyFreeformRuleset.cs
+++ b/Templates/Rulesets/ruleset-empty/osu.Game.Rulesets.EmptyFreeform/EmptyFreeformRuleset.cs
@@ -25,6 +25,8 @@ public class EmptyFreeformRuleset : Ruleset
     {
         public override string Description => "a very emptyfreeformruleset ruleset";

+        public override string RulesetAPIVersionSupported => CURRENT_RULESET_API_VERSION;
+
         public override DrawableRuleset CreateDrawableRulesetWith(IBeatmap beatmap, IReadOnlyList<Mod> mods = null) =>
             new DrawableEmptyFreeformRuleset(this, beatmap, mods);

diff --git a/osu.Game/Rulesets/Ruleset.cs b/osu.Game/Rulesets/Ruleset.cs
index 63f5906f46..1a39bb55ac 100644
--- a/osu.Game/Rulesets/Ruleset.cs
+++ b/osu.Game/Rulesets/Ruleset.cs
@@ -49,7 +49,7 @@ public abstract class Ruleset
         /// 2022.205.0   FramedReplayInputHandler.CollectPendingInputs renamed to FramedReplayHandler.CollectReplayInputs.
         /// 2022.822.0   All strings return values have been converted to LocalisableString to allow for localisation support.
         /// </summary>
-        public const string CURRENT_RULESET_API_VERSION = "2022.822.0";
+        public const string CURRENT_RULESET_API_VERSION = "2022.822.1";

         /// <summary>
         /// Define the ruleset API version supported by this ruleset.
diff --git a/osu.Game/Rulesets/RulesetStore.cs b/osu.Game/Rulesets/RulesetStore.cs
index fdbcd0ed1e..ed4f52b447 100644
--- a/osu.Game/Rulesets/RulesetStore.cs
+++ b/osu.Game/Rulesets/RulesetStore.cs
@@ -27,13 +27,13 @@ protected RulesetStore(Storage? storage = null)
         {
             // On android in release configuration assemblies are loaded from the apk directly into memory.
             // We cannot read assemblies from cwd, so should check loaded assemblies instead.
-            loadFromAppDomain();
+            // loadFromAppDomain();

             // This null check prevents Android from attempting to load the rulesets from disk,
             // as the underlying path "AppContext.BaseDirectory", despite being non-nullable, it returns null on android.
             // See https://github.com/xamarin/xamarin-android/issues/3489.
-            if (RuntimeInfo.StartupDirectory != null)
-                loadFromDisk();
+            // if (RuntimeInfo.StartupDirectory != null)
+            //     loadFromDisk();

             // the event handler contains code for resolving dependency on the game assembly for rulesets located outside the base game directory.
             // It needs to be attached to the assembly lookup event before the actual call to loadUserRulesets() else rulesets located out of the base game directory will fail
```